### PR TITLE
Build and run all tests found by the glob.

### PR DESCRIPTION
**Pull request by wohe**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -86,6 +86,14 @@ list(APPEND ALL_SRCS ${ALL_PROTO_SRCS} ${ALL_PROTO_HDRS})
 add_library(${PROJECT_NAME} ${ALL_SRCS})
 add_subdirectory("cartographer")
 
+foreach(ABS_FIL ${ALL_TESTS})
+  file(RELATIVE_PATH REL_FIL ${PROJECT_SOURCE_DIR} ${ABS_FIL})
+  get_filename_component(DIR ${REL_FIL} DIRECTORY)
+  get_filename_component(FIL_WE ${REL_FIL} NAME_WE)
+  message(INFO google_test "${DIR}/${FIL_WE}" ${ABS_FIL})
```
Comment (SirVer) [inline CMakeLists.txt]: debug cruft?

**Comment by wohe** [comment]
This fixes #151.

**Comment by SirVer** [comment]
Testing github linking. This "fixes" #151.

**Comment by SirVer** [comment]
This fixes #151.

**Comment by BrannonKing** [comment]
This change doesn't seem to work for me. I get this compilation error:

```
[ 78%] Built target cartographer/sensor/range_data_test
[ 79%] Built target cartographer/mapping_3d/kalman_local_trajectory_builder_test
[ 80%] Built target cartographer/mapping_3d/motion_filter_test
[ 81%] Built target cartographer/mapping_3d/scan_matching/precomputation_grid_test
[ 82%] Built target cartographer/mapping_3d/hybrid_grid_test
[ 83%] Built target cartographer/sensor/ordered_multi_queue_test
[ 83%] Built target build_doc
[ 84%] Built target cartographer/mapping_2d/scan_matching/correlative_scan_matcher_test
[ 85%] Built target cartographer/mapping_3d/scan_matching/real_time_correlative_scan_matcher_test
[ 86%] Built target cartographer/transform/transform_interpolation_buffer_test
[ 86%] Linking CXX executable cartographer/mapping_3d/sparse_pose_graph/optimization_problem_test
[ 86%] Built target cartographer/common/math_test
[ 87%] Built target cartographer/mapping_2d/scan_matching/real_time_correlative_scan_matcher_test
[ 87%] Built target cartographer/mapping/probability_values_test
Scanning dependencies of target fast_correlative_scan_matcher_test
/usr/bin/ld: cannot open output file cartographer/mapping_3d/sparse_pose_graph/optimization_problem_test: No such file or directory
collect2: error: ld returned 1 exit status
CMakeFiles/cartographer/mapping_3d/sparse_pose_graph/optimization_problem_test.dir/build.make:124: recipe for target 'cartographer/mapping_3d/sparse_pose_graph/optimization_problem_test' failed
make[2]: *** [cartographer/mapping_3d/sparse_pose_graph/optimization_problem_test] Error 1
CMakeFiles/Makefile2:1092: recipe for target 'CMakeFiles/cartographer/mapping_3d/sparse_pose_graph/optimization_problem_test.dir/all' failed
make[1]: *** [CMakeFiles/cartographer/mapping_3d/sparse_pose_graph/optimization_problem_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 87%] Linking CXX executable cartographer/io/proto_stream_test
[ 88%] Built target cartographer/mapping_2d/probability_grid_test
Scanning dependencies of target interpolated_grid_test
[ 88%] Built target cartographer/mapping_2d/range_data_inserter_test
[ 89%] Built target cartographer/common/blocking_queue_test
/usr/bin/ld: cannot open output file cartographer/io/proto_stream_test: No such file or directory
collect2: error: ld returned 1 exit status
CMakeFiles/cartographer/io/proto_stream_test.dir/build.make:124: recipe for target 'cartographer/io/proto_stream_test' failed
make[2]: *** [cartographer/io/proto_stream_test] Error 1
CMakeFiles/Makefile2:1206: recipe for target 'CMakeFiles/cartographer/io/proto_stream_test.dir/all' failed
make[1]: *** [CMakeFiles/cartographer/io/proto_stream_test.dir/all] Error 2
[ 90%] Built target cartographer/mapping/submaps_test
[ 90%] Building CXX object CMakeFiles/cartographer/mapping_2d/scan_matching/fast_correlative_scan_matcher_test.dir/cartographer/mapping_2d/scan_matching/fast_correlative_scan_matcher_test.cc.o
[ 91%] Building CXX object CMakeFiles/cartographer/mapping_3d/scan_matching/interpolated_grid_test.dir/cartographer/mapping_3d/scan_matching/interpolated_grid_test.cc.o
[ 91%] Linking CXX executable cartographer/mapping_3d/scan_matching/interpolated_grid_test
[ 91%] Built target cartographer/mapping_3d/scan_matching/interpolated_grid_test
[ 92%] Linking CXX executable cartographer/mapping_2d/scan_matching/fast_correlative_scan_matcher_test
[ 92%] Built target cartographer/mapping_2d/scan_matching/fast_correlative_scan_matcher_test
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by SirVer** [comment]
@BrannonKing can you try building in a new empty catkin_workspace from scratch?

**Comment by BrannonKing** [comment]
The error occurs even in a clean workspace, but I'm compiling with ament, not catkin. You can see here that the makefile is not changed, though: https://github.com/asirobots/cartographer/

**Comment by SirVer** [comment]
@BrannonKing we cannot repro this error on our CI and also not locally. I think you need to try to repro our build instructions and build with catkin and see if that works and investigate from there.

**Pull request merged**